lib/commands.js

```diff
@@ -17,18 +17,14 @@ var changeFunctionName = (function () {
     }
 }());
 
-// TODO: Rewrite this including the invidual commands into a Commands class
-// that provided a functionality to add new commands to the client
-
-commands.list.forEach(function (command) {
-
+var addCommand = function (command) {
     // Some rare Redis commands use special characters in their command name
     // Convert those to a underscore to prevent using invalid function names
     var commandName = command.replace(/(?:^([0-9])|[^a-zA-Z0-9_$])/g, '_$1');
 
     // Do not override existing functions
-    if (!RedisClient.prototype[command]) {
-        RedisClient.prototype[command.toUpperCase()] = RedisClient.prototype[command] = function () {
+    if (!RedisClient.prototype[commandName]) {
+        RedisClient.prototype[commandName.toUpperCase()] = RedisClient.prototype[commandName] = function () {
```

Comment:
So the reason is: Redis itself has a command with the name `restore-asking`. And if you want to use this command anyone should be able to call the function with exactly this name, not a different one. But to have a function name in a possible stack trace the name is added to the function itself. And if it contains illegal function name characters those characters are converted to a underscore.
Another solution would have been to special handle this command and to document it but this is something I'm not in favor of.